src/components/RightPane/CoursePane/SearchForm/SearchForm.tsx

```diff
@@ -48,7 +48,33 @@ const styles: Styles<Theme, object> = {
 const SearchForm = (props: { classes: ClassNameMap; toggleSearch: () => void }) => {
     const { classes, toggleSearch } = props;
 
-    const [showLegacySearch, setShowLegacySearch] = useState(false);
+    const urlParamValue1 =
+        new URLSearchParams(window.location.search).get('courseCode') != '' &&
+        new URLSearchParams(window.location.search).get('courseCode') != null;
+    const urlParamValue2 =
+        new URLSearchParams(window.location.search).get('courseNumber') != '' &&
+        new URLSearchParams(window.location.search).get('courseNumber') != null;
+    const urlParamValue3 =
+        new URLSearchParams(window.location.search).get('deptLabel') != '' &&
+        new URLSearchParams(window.location.search).get('deptLabel') != null;
+    const urlParamValue4 =
+        new URLSearchParams(window.location.search).get('deptValue') != '' &&
+        new URLSearchParams(window.location.search).get('deptValue') != null;
+    const urlParamValue5 =
+        new URLSearchParams(window.location.search).get('GE') != '' &&
+        new URLSearchParams(window.location.search).get('GE') != null;
+    const urlParamValue6 =
+        new URLSearchParams(window.location.search).get('term') != '' &&
+        new URLSearchParams(window.location.search).get('term') != null;
+    const allParamValues = [
+        urlParamValue1,
+        urlParamValue2,
+        urlParamValue3,
+        urlParamValue4,
+        urlParamValue5,
+        urlParamValue6,
+    ];
```

Comment:
Only calculate the search params once. Also, if these params are only used once, you can just inline it into the array, or inline a conditional. The latter is more similar to what you did initially lol; just some ideas. Definitely only use `new URLSearchParams` once tho.

```suggestion
    const search = new URLSearchParams(window.location.search)
    const urlParamValue1 = search.get('courseCode') != '' && search && != null;
    const urlParamValue2 = search != '' && search.get('courseNumber') != null;
    const urlParamValue3 = search.get('deptLabel') != '' &&  search.get('deptLabel') != null;
    const urlParamValue4 = search.get('deptValue') != '' && search.get('deptValue') != null;
    const urlParamValue5 = search.get('GE') != '' && search.get('GE') != null;
    const urlParamValue6 = search.get('term') != '' && search.get('term') != null;
    const allParamValues = [
        urlParamValue1,
        urlParamValue2,
        urlParamValue3,
        urlParamValue4,
        urlParamValue5,
        urlParamValue6,
    ];
    
    // alternatively
   const isLegacySearch = 
    (search.get('courseCode') != '' && search && != null) ||
    (search != '' && search.get('courseNumber') != null) ...
    
     const [showLegacySearch, setShowLegacySearch] = useState(isLegacySearch)
```